swri_dbw_interface/doc/package.md

```diff
@@ -0,0 +1,366 @@
+interface that the rest of our system interacts with.  This layer may
+be anything from simple translations between message types to complex
+closed loop controllers.  It may consist of one node or many nodes.
+The important aspect is a consistent interface presented to the higher
+level autonomy system, not how it is implemented.
+
+This interface has evolved against actuated gas/brake
+Ackermann-steered vehicles (as opposed to tracked vehicles or vehicles
+that provide intrinsic speed control).  It is expected that the
+interface will be extended to handle other types of vehicles as they
+are encountered.
+
+Due to the combined unique needs of projects and variations in
+drive-by-wire capabilities and implementations, it is expected that
+variations and adaptations may be required for different
+circumastances.  In those cases, any variations should be documented
+along with the implementation.
+
+As new features are required, it is recommended that they be developed
+for the specific project according to the conventions below, tested
+and refined in an actual use case, and then, if they are generic
+enough for other applications, merged into these specifications once
+they are mature.
+
+
+## Conventions [⬆](#swri_dbw_interface)
+
+We have developed many conventions that are followed to have as much
+internal consistency as possible.
+
+1. Every axis of control should have a command topic, used by the
+   higher autonomy to control the hardware, and a feedback topic, used
+   by the higher autonomy to determine the current state of the axis
+   (or the predicted state if no actual feedback is available).
+
+1. Command topics should have topic names with the postfix "_command".
+
+1. Feedback topics should have topic names with the postfix "_feedback".
+
+1. Other topics should be given names that make it obvious whether
+   they are an input to or output from the DBW system.
+
+1. Prefer to use strings (with constants) over integer enumerations so
+   they are easy to introspect using "rostopic echo".
+
+1. For axes that change infrequently (e.g, transmission), make the
+   feedback topic latched.  Publish the state when it changes, and at
+   a fixed slow rate (e.g. 1Hz) for easier usage with bag files.  This
+   rate can be a configurable parameter.  This specification will
+   label topics that follow this convention as SLOW/ONCHANGE.
+
+1. Prefer topics over services for easier usage with bag files.
+
+1. If a DBW implementation provides feedback but no control of an
+   axis, it is acceptable to either subscribe but ignore message on
+   tha command topic, or not subscribe to the topic at all.
+
+1. Use the ROS console macros to provide useful feedback for
+   developers and troubleshooters, but be careful not spam the output.
+
+
+## Namespace [⬆](#swri_dbw_interface)
+
+By convention, all topics related to the drive-by-wire system are in
+the "/vehicle_interface" namespace.
```

Comment:
Use of this namespace is one reason why I preferred the name swri_vehicle_interface to swri_dbw_interface.  Is there any value to using /dbw_interface as the namespace instead?  In any case, I think it would be beneficial to segregate the swri_dbw_interface topics from any native dbw ROS topics by namespace... perhaps /dbw_interface would be a good place to put those.

Reply:
I prefer /vehicle/ or /vehicle_interface/ to /dbw_interface/

Reply:
I agree, and I prefer /vehicle_interface to /vehicle. Does it make sense to you to segregate native DBW messages in a different namespace? I know we mix them on some (all?) platforms, and I find it difficult to distinguish which are which at a glance. Also, there's the (small) possibility that naming conventions may conflict.